src/Components/Publishing/Display/DisplayAd.tsx

```diff
@@ -30,84 +37,98 @@ export interface DisplayAdContainerProps extends FlexProps {
 
 GPT.syncCorrelator(true)
 
-export const DisplayAd: SFC<DisplayAdProps> = props => {
-  const {
-    adDimension,
-    adUnit,
-    targetingData,
-    articleSlug,
-    ...otherProps
-  } = props
-  const [width, height] = adDimension.split("x").map(a => parseInt(a))
-  const [isAdEmpty, setAdEmpty] = useState(null)
-  const isMobileLeaderboardAd = is300x50AdUnit(adDimension)
-  const { trackEvent } = useTracking()
-
-  useEffect(() => {
-    trackEvent({
-      action_type: AnalyticsSchema.ActionType.Impression,
-      context_page: AnalyticsSchema.PageName.ArticlePage,
-      context_module: AnalyticsSchema.ContextModule.AdServer,
-      context_page_owner_id: props.targetingData.post_id,
-      context_page_owner_slug: props.articleSlug,
-      context_page_owner_type: AnalyticsSchema.OwnerType.Article,
-    })
-  }, [])
+@track(null, {
+  dispatch: data => Events.postEvent(data),
+})
+export class DisplayAd extends React.Component<DisplayAdProps> {
+  state = {
+    isAdEmpty: null,
+  }
+  @track<DisplayAdProps>(props => ({
+    action_type: AnalyticsSchema.ActionType.Impression,
+    context_page: AnalyticsSchema.PageName.ArticlePage,
```

Comment:
>  we reverted to using a class component because it was difficult to get Waypoints to work with functional components.

This is correct!

Reply:
> The future of Artsy's codebase is based on functional components, not classes. It would benefit us long-term to figure out how to use Waypoints with functional components.

Is this something we can add a ticket to the backlog for and perhaps do a mini spike for in a future sprint?

Reply:
The other changes were made and its all green. I'm going to merge so @abhitip can test this on Staging!